ftfy/fixes.py

```diff
@@ -230,20 +261,28 @@ def apply_plan(text, plan):
     """
     Apply a plan for fixing the encoding of text.
 
-    The plan is a list of tuples of the form (operation, encoding), where
-    `operation` is either 'encode' or 'decode', and `encoding` is an encoding
-    name such as 'utf-8' or 'latin-1'.
+    The plan is a list of tuples of the form (operation, encoding, cost):
 
-    Because only text can be encoded, and only bytes can be decoded, the plan
-    should alternate 'encode' and 'decode' steps, or else this function will
-    encounter an error.
+    - `operation` is 'encode' if it turns a string into bytes, 'decode' if it
+      turns bytes into a string, and 'transcode' if it keeps the type the same.
+    - `encoding` is the name of the encoding to use, such as 'utf-8' or
+      'latin-1', or the function name in the case of 'transcode'.
+    - `cost` is a penalty to apply to the score of the resulting text, when
+      performing a dubious operation that requires a lot of evidence.
```

Comment:
The description of `cost` is kind of confusing, because cost is pretty much an internal thing that the user doesn't usually need to know about.  This function doesn't actually *use* the costs, but this docstring makes it sound like it might use them, or it's unclear who's doing the applying of the penalty.  Maybe you could explicitly say that `cost` is a thing used by other users of plans, namely `fix_encoding_and_explain`, which has to decide *which* plan to use.